File: 0724-find-pivot-index/0724-find-pivot-index.go
func pivotIndex(nums []int) int {
    total := 0
    for _, val := range nums {
        total = total + val
    }

    left := 0
    for i := 0; i < len(nums); i++ {
        right := total - left - nums[i]
        if right == left {
            return i
        }
        left = left + nums[i]
    }

    return -1
}

File: 0724-find-pivot-index/0724-find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int total = accumulate(nums.begin(), nums.end(), 0);
        int left = 0;

        for (int i = 0; i < nums.size(); i++) {
            int right = total - left - nums[i];
            if (right == left) {
                return i;
            }
            left = left + nums[i];
        }

        return -1;
    }
};